### DISCUSSION TITLE
rootless podman cannot build with docker-compose build

### DISCUSSION THREAD

**Discussion by amyfou** [containers/podman#16338] [General]
trying to build images rootless using podman - build fails with error 

`DOCKER_BUILDKIT=0 docker-compose -f docker-compose-prod.yml build`

```
ERRO[0000] Can't add file /home/usernm/localdirectorypath/package-lock.json to tar: io: read/write on closed pipe 
ERRO[0000] Can't close tar writer: io: read/write on closed pipe 
ERRO[0001] Can't add file /home/usernm/localdirectorypath/package-lock.json to tar: io: read/write on closed pipe 
ERRO[0001] Can't close tar writer: io: read/write on closed pipe 
3 errors occurred:
	* error during connect: Post "http://unix:2375/run/podman/podman.sock/v1.24/build?buildargs=%7B%7D&cachefrom=null&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile.dev&labels=%7B%22com.docker.compose.image.builder%22%3A%22classic%22%2C%22com.docker.compose.project%22%3A%22colrc-v2%22%2C%22com.docker.compose.service%22%3A%22backend%22%2C%22com.docker.compose.version%22%3A%222.11.1%22%2C%22edu.arizona.colrc.description%22%3A%22API+for+COLRC+v2%22%7D&memory=0&memswap=0&networkmode=&rm=1&shmsize=0&t=colrc-v2-backend&target=&ulimits=null&version=1": dial tcp: lookup unix on 108.61.10.10:53: no such host
	* checking context: can't stat '/home/usernm/localdirectorypath'
	* error during connect: Post "http://unix:2375/run/podman/podman.sock/v1.24/build?buildargs=%7B%7D&cachefrom=null&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile.dev&labels=%7B%22com.docker.compose.image.builder%22%3A%22classic%22%2C%22com.docker.compose.project%22%3A%22colrc-v2%22%2C%22com.docker.compose.service%22%3A%22file_upload%22%2C%22com.docker.compose.version%22%3A%222.11.1%22%2C%22edu.arizona.colrc.description%22%3A%22File+upload+processor+for+COLRC+v2%22%7D&memory=0&memswap=0&networkmode=&rm=1&shmsize=0&t=colrc-v2-file_upload&target=&ulimits=null&version=1": dial tcp: lookup unix on 108.61.10.10:53: no such host
```
build works with sudo.

**Comment by TomaszGasior** [comment]
What is your `DOCKER_HOST`? 

The output suggests you are trying to connect from docker-compose to podman deamon running as root (podman daemon started system-wide). Let's try to use 
```
export DOCKER_HOST=unix://$XDG_RUNTIME_DIR/podman/podman.sock
```
and enable user-wide podman daemon by
```
systemctl enable --now --user podman.socket
```

Don't use sudo for any command.

**Reply by amyfou** [reply]
Thank you - my DOCKER_HOST was:  `unix:/run/podman/podman.sock`

running the commands you've given here has fixed the problem.  I really appreciate your help!

**Reply by amyfou** [reply]
UPDATE:  it was working.  We removed all images, pods and volumes, then tried to rebuild using the commands you've given above; and then the same build script that worked last time - but now we're seeing this error:

```Cannot connect to the Docker daemon at unix:///run/user/1000/podman/podman. Is the docker daemon running?```

Can't find any examples of resolution of this error online, do you know what's up?

**Reply by TomaszGasior** [reply]
Do you have podman deamon running? (Just like in error message). What's in `systemctl status --user podman.socket` (no sudo)?

Maybe you need to `systemctl enable --now --user podman.socket` (no sudo)?

**Reply by TomaszGasior** [reply]
Also, you may have incorrect value of DOCKER_HOST since the error message says "Cannot connect to the Docker daemon at **unix:///run/user/1000/podman/podman**". There is no ".sock" at the end.

**Comment by amyfou** [comment]
Thanks again - here's the output of systemctl status --user podman.socket:

```systemctl status --user podman.socket
● podman.socket - Podman API Socket
     Loaded: loaded (/usr/lib/systemd/user/podman.socket; enabled; vendor prese>
     Active: active (listening) since Tue 2022-11-01 22:08:06 UTC; 1h 20min ago
   Triggers: ● podman.service
       Docs: man:podman-system-service(1)
     Listen: /run/user/1000/podman/podman.sock (Stream)
      Tasks: 0 (limit: 9507)
     Memory: 0B
        CPU: 0
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/podman.soc>

Nov 01 22:08:06 colrc systemd[3095196]: Listening on Podman API Socket.```

**Reply by amyfou** [reply]
and our $DOCKER_HOST

```unix:///run/user/1000/podman/podman```

**Reply by amyfou** [reply]
ugh, you're right no .sock

omg so sorry for the extra pester.  Thank you again!

**Comment by flouthoc** [comment]
Looks already solved hence closing this.
